File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 2/12/2026, 3:45:37 PM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = float('inf')
        profit = 0
        for sell in prices:
            if sell < buy:
                buy = sell
            elif sell - buy > profit:
                profit = sell - buy
        return profit

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 2/12/2026, 3:45:34 PM
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minCost = INT_MAX;
        int maxProfit = 0;
        for (int currentPrice : prices) {
            if (currentPrice < minCost) {
                minCost = currentPrice;
            }
            else {
                maxProfit = max(maxProfit, currentPrice - minCost);
            }
        }
        return maxProfit;
    }
};